openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -1391,9 +1388,7 @@ impl UhProcessor<'_, TdxBacked> {
         // run code on the next exit to clear it. If we miss this opportunity,
         // we will probably double-inject the interruption, wreaking havoc.
         let offload_enabled = next_vtl == GuestVtl::Vtl0
-            && self.backing.vtls[next_vtl]
-                .secondary_processor_controls
-                .virtual_interrupt_delivery()
+            && self.backing.vtls[next_vtl].offload_apic
```

Comment:
In fact, even this check is unnecessary, because `can_offload_irr` won't be true if this is false.